test/src/e2e_vm_tests/test_programs/supertraits/src/main.sw

```diff
@@ -0,0 +1,179 @@
+script;
+use core::*;
+use core::ops::Ord;
+
+struct Rgb {
+    red: u64,
+    green: u64,
+    blue: u64,
+}
+
+trait ColorBase1 {
+    fn rgb1(self) -> Rgb;
+} {
+    fn rgb1_wrapper(self) -> Rgb {
+        self.rgb1()
+    } 
+} 
+
+trait ColorBase2 {
+    fn rgb2(self) -> Rgb;
+} {
+    fn rgb2_wrapper(self) -> Rgb {
+        self.rgb2()
+    } 
+} 
+
+
+trait ColorBase3 : ColorBase1 {}
+
+trait ColorBase4 : ColorBase2 {}
+
+trait Color : ColorBase3 + ColorBase4 {
+    fn rgb(self) -> Rgb;
+} {
+    fn rgb_wrapper(self) -> Rgb {
+        self.rgb()
+    } 
+} 
+
+enum PrimaryColor {
+    Red: (),
+    Green: (),
+    Blue: (),
+}
+
+impl core::ops::Ord for PrimaryColor {
+    fn lt(self, other: Self) -> bool {
+        asm(r1: self, r2: other, r3) {
+            lt r3 r1 r2;
+            r3: bool
+        }
+    }
+    fn gt(self, other: Self) -> bool {
+        asm(r1: self, r2: other, r3) {
+            gt r3 r1 r2;
+            r3: bool
+        }
+    }
+    fn eq(self, other: Self) -> bool {
+        asm(r1: self, r2: other, r3) {
+            eq r3 r1 r2;
+            r3: bool
+        }
+    }
+}
+
```

Comment:
I'm pretty sure these aren't correct -- you've copied them from every other `Ord` impl we have in std, except since `self` is an enum, the values for `r1`and `r2` in the ASM blocks are pointers.  And comparing colours is a bit strange -- I guess you could order them  as red being small and blue being big, or something.

I'm guessing this is here just to make sure regular traits still compile OK?  But the actual impl fns should probably be more correct.  Like `Red` < `Green` < `Blue` or something.

Reply:
I actually copied most of this test from another test, but I agree - the test could be drastically improved. I will address both of your comments and update the PR accordingly. Thanks!